src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java

```diff
@@ -80,16 +85,40 @@ private static SymbolLookup makeSystemLookup() {
         }
     }
 
+    /* Obtain the address of the inlined function list by loading the specified library. */
+    private static MemorySegment getInlinedFunctListAddr() {
+         MemorySegment funcsAddr;
+         long FuncSymbolsSize = OperatingSystem.isAix()
+                 ? AixFuncSymbols.LAYOUT.byteSize()
+                 : ZosFuncSymbols.LAYOUT.byteSize();
+
+        /* Directly load the specified library only once with the inlined libc functions
+         * missing in the default library.
+         */
+        SymbolLookup funcsLibLookup = libLookup(libs -> libs.load(jdkLibraryPath("syslookup")));
+
+        /* Perform a FFI downcall to obtain the address of the inlined function list. */
+        MemorySegment funcs_addr = funcsLibLookup.find("funcs_addr").orElseThrow();
+        MethodHandle handle = Linker.nativeLinker().downcallHandle(funcs_addr, FunctionDescriptor.of(ADDRESS));
+        try {
+            funcsAddr = ((MemorySegment)handle.invoke()).reinterpret(FuncSymbolsSize);
+        } catch (Throwable e) {
+            throw new InternalError(e);
+        }
+
+        return funcsAddr;
```

Comment:
Suggest:
* moving the declaration of `funcsAddr` to immediately before it gets assigned
* delaying the call to `byteSize()`
* pulling `reinterpret()` outside the `try` block

```java
        /* Directly load the specified library only once with the inlined libc functions
         * missing in the default library.
         */
        SymbolLookup funcsLibLookup = libLookup(libs -> libs.load(jdkLibraryPath("syslookup")));

        /* Perform a FFI downcall to obtain the address of the inlined function list. */
        MemorySegment funcs_addr = funcsLibLookup.find("funcs_addr").orElseThrow();
        MethodHandle handle = Linker.nativeLinker().downcallHandle(funcs_addr, FunctionDescriptor.of(ADDRESS));
        MemorySegment funcsAddr;
        try {
            funcsAddr = (MemorySegment) handle.invoke();
        } catch (Throwable e) {
            throw new InternalError(e);
        }

        var funcsLayout = OperatingSystem.isAix() ? AixFuncSymbols.LAYOUT : ZosFuncSymbols.LAYOUT;

        return funcsAddr.reinterpret(funcsLayout.byteSize());
```

Reply:
Agreed and updated the code against the suggestion above.